test/model.test.js

```diff
@@ -112,6 +112,19 @@ describe('loopback:model generator', function() {
       });
   });
 
+  it('honors the model name in arg', function() {
+    return helpers.run(path.join(__dirname, '../model'))
+      .cd(SANDBOX)
+      .withArguments(['Product'])
+      .withPrompts({
+        dataSource: 'db',
+        base: 'PersistedModel',
+      }).then(function() {
+        var product = readProductJsonSync();
+        expect(product).to.have.property('base', 'PersistedModel');
```

Comment:
I find this slightly confusing - how is the base property related to the model name provided via CLI arguments? Should you perhaps check that `name` property was set to `Product` instead?

Reply:
> The test before it sets the base to CustomModel, therefore my test reset it to PersistedModel and assert accordingly.

Could you please add a comment to explain this?

In general, tests should be independent on each other and don't rely on side effects.

> but even the test fail this assertion would still be true :)

Isn't the same true for your current test in case only this single test is executed and all other tests are skipped? Either via `it.only` or in CLI via `mocha -g "loopback:model generator.*honors the model name in arg"`.

I think you should explicitly set a different base model in this test, one that's not the default and also not used by any other tests around. Perhaps `Model` may work? Then verify that this explicit value was written to the file.

Reply:
@bajtos Hmm I see your concern. I rewrote the test, hopefully it makes more sense.